src/NLog/Targets/MailTarget.cs

```diff
@@ -408,10 +423,10 @@ private void ConfigureMailClient(LogEventInfo lastEvent, ISmtpClient client)
 
         private void CheckRequiredParameters()
         {
-            if (!this.UseSystemNetMailSettings && this.SmtpServer == null)
+            if (!this.UseSystemNetMailSettings && this.SmtpServer == null && string.IsNullOrEmpty(this.PickupDirectoryLocation))
```

Comment:
I think this is wrong, but with another reason.

If `UseSystemNetMailSettings = false`, only `SmtpServer` OR `PickupDirectoryLocation` is required, not both!

Because I can send with SMTP (only smtp needed), or I can send to pickup (only location is needed).

Can we change this, and also, change it to 2 `if` statements to keep it more clear?

I would be nice if we also unit test this case (that not both are required)